### PR TITLE
Linux loader integration

### PR DESCRIPTION
**Pull request by AlexandruCihodaru**
# Reason for This PR

Replaced kernel crate with rust-vmm/linux-loader
See https://github.com/firecracker-microvm/firecracker/issues/1182.

## Description of Changes

Added github dependency for rust-vmm/linux-loader
Deleted kernel crate from Firecracker
Use linux-loader to write bootparams
Integrated rust-vmm/linux-loader in vmm crate

As for right now the snapshot tests will not pass since the rust compiler uses upstream implementations of traits from vm-memory.

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any newly added `unsafe` code is properly documented.
- [x] Any API changes are reflected in `firecracker/swagger.yaml`.
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.

### CODE REVIEW

**Comment by andreeaflorescu** [comment]
We just yanked v0.2.0 of linux-loader. Before merging the PR, please make sure to use the latest available version v0.3.0.

**Review by dianpopa — COMMENTED** [review]
The changes where you make use of the boot_params defined in the linux-loader from rust-vmm could be placed in another commit for better readability.

**Comment by dianpopa** [comment]
This is blocked on release v0.6 of [rust-vmm/vm-memory](https://github.com/rust-vmm/vm-memory/pull/148).

**Comment by alindima** [comment]
This should also be rebased on top of https://github.com/firecracker-microvm/firecracker/pull/2661, once it's merged. (It's also blocked on vm-memory v0.6 release)

**Comment by dianpopa** [comment]
After #2661 gets merged, I guess we can also resume work on this. Is there anything else we are depending on here @AlexandruCihodaru?

**Comment by andreeaflorescu** [comment]
We just published v0.4.0 which should unblock this PR on the consumption of kernel loading functionality.

File: src/vmm/src/builder.rs
Comment (alindima) [inline src/vmm/src/builder.rs:571]: these inline cfg's are hard to parse visually. Maybe add conditional compilation to the entire variable definition?

File: src/vmm/src/builder.rs
Comment (alindima) [inline src/vmm/src/builder.rs:991]: well the default is much smaller (100 chars), so why don't we allocate the actual size of it?
If not, we may be allocating 65536 unused bytes (on x86)

**Review by alindima — COMMENTED** [review]
Looks good! Only a couple of nits

File: src/vmm/src/builder.rs
Comment (AlexandruCihodaru) [reply src/vmm/src/builder.rs:991]: we append to it so the size will actually grow, that is why we make it bigger than default value

**Review by alindima — APPROVED** [review]

File: src/vmm/src/vstate/vcpu/mod.rs
Comment (alindima) [inline src/vmm/src/vstate/vcpu/mod.rs:885]: missed to remove this `cfg`

File: src/vmm/src/resources.rs
Comment (alindima) [inline src/vmm/src/resources.rs:899]: it's best to keep the `assert_eq`

**Review by alindima — APPROVED** [review]

File: src/vmm/src/vstate/vcpu/mod.rs
Comment (dianpopa) [inline src/vmm/src/vstate/vcpu/mod.rs:874]: There is already `test_elf` and `test_pe` in `vmm/src/utilities/mock_resources`. Please try to use those since they are already in the same crate as this test. If all is good, we will be able to delete the once you relocated in `tests/resources`.

**Review by alindima — APPROVED** [review]

**Review by dianpopa — APPROVED** [review]

**Pull request merged**
